pkg/minishift/openshift/registry.go

```diff
@@ -21,10 +21,53 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/minishift/minishift/pkg/minishift/addon/manager"
 	instanceState "github.com/minishift/minishift/pkg/minishift/config"
 )
 
-func GetDockerRegistryInfo() (string, error) {
+func GetDockerRegistryInfo(addOnManager *manager.AddOnManager) (string, error) {
+	var registryInfo string
+	var err error
+	if isRegistryRouteEnable(addOnManager) {
+		registryInfo, err = fetchRegistryRoute()
+	} else {
+		registryInfo, err = fetchRegistryService()
+	}
+	return registryInfo, err
+}
+
+func isRegistryRouteEnable(addOnManager *manager.AddOnManager) bool {
+	registryRoute := addOnManager.Get("registry-route")
+	if registryRoute == nil || !registryRoute.IsEnabled() {
+		return false
+	}
+	namespace := "default"
+	cmdArgText := fmt.Sprintf("get route docker-registry -n %s --config=%s", namespace, systemKubeConfigPath)
```

Comment:
@hferentschik yes this is just an additional check if addon works and route is enabled.